### ISSUE TITLE
[windows] UNC path not supported by mamba, supported by conda

### ISSUE THREAD

**Issue by adament** [mamba-org/mamba#733]
In conda on windows it is possible to refer to a channel on a windows share using a UNC path using
```yaml
channels:
- file:////hostname/share
- file://\\hostname\share
```
Curl supports using the `file:////hostname/share` URL to refer to a UNC path. However it seems that mamba replaces the quadruple slash with a double slash before passing it on to curl as `file://hostname/share`. Would it be possible to keep the quadruple slash and pass it to libcurl? We are using conda in a Windows environment for internal package repositories and UNC paths and fileshares provides a simple way to apply access control to the repositories.

I have tested with:
```
mamba 0.7.14
conda 4.9.2
```

**Comment by wolfv** [comment]
A PR would be very welcome!

**Comment by adament** [comment]
I think I have identified the issue to [line 463](https://github.com/mamba-org/mamba/blob/243bf801e99fc2215f88fd87c66574aeb85a52e9/src/channel.cpp#L463) in the `fix_win_path` function in `channel.cpp`. However given the [test](https://github.com/mamba-org/mamba/blob/243bf801e99fc2215f88fd87c66574aeb85a52e9/test/test_channel.cpp#L28) it seems the like the incompatibility with conda was intentional.

I will try simply removing that line and changing the test and see if that fixes my issue, if so I will make a pull request.

**Comment by adament** [comment]
Turns out this was not correct. It seems that the current implementation of fix_win_path just matches win_path_backout in conda.common.path so when calling conda.core.index.calculate_channel_urls the result has stripped the initial // from the UNC url as well. I do not know where conda makes that back into a UNC path again.

**Comment by adament** [comment]
My current guess is that conda fixes the problem created by stripping the UNC prefix by reintroducing it again in url_to_path in conda.common.path.

**Comment by wolfv** [comment]
if there is absolutely no cleaner way, then yeah, I guess we have to do it.

btw. just to make sure: can you check that this URL format works with curl? (`curl file:////.... -o test.json`) so that we know we won't hit problems in curl later on.

**Comment by adament** [comment]
This should be closed with the merge of #749

**Comment by wolfv** [comment]
Thanks @adament !
